src/stories/TextInput.tsx

```diff
@@ -0,0 +1,161 @@
+
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+
+  const regex = new RegExp(validationRegex);
+  const isValid = regex.test(realValue);
+  const validCharacters = /^[^\s\x00-\x1F\x7F-\x9F\u2028\u2029]$/;
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (e.key === "Backspace") {
+      if (selectionStart === selectionEnd) {
+        setRealValue(
+          realValue.slice(0, selectionStart - 1) +
+            realValue.slice(selectionEnd, realValue.length)
+        );
+      } else {
+        setRealValue(
+          realValue.slice(0, selectionStart) +
+            realValue.slice(selectionEnd, realValue.length)
+        );
+      }
+      if (textareaRef.current) {
+        textareaRef.current.setSelectionRange(0, 0);
+      }
+      //flickers
+    } else if (validCharacters.test(e.key)) {
+      setRealValue(
+        realValue.slice(0, selectionStart) +
+          e.key +
+          realValue.slice(selectionEnd, realValue.length)
+      );
+    }
+  };
+
+  const handleSelectionChange = (e: React.MouseEvent<HTMLTextAreaElement>) => {
+    setSelectionStart(e.currentTarget.selectionStart);
+    setSelectionEnd(e.currentTarget.selectionEnd);
+  };
+
+  useEffect(() => {
+    setDisplayValue(isPassword ? "•".repeat(realValue.length) : realValue);
+    console.log(realValue, displayValue);
+  }, [realValue, isPassword]);
+
+  return (
+    <div style={{ display: "flex" }} className="storybook-textInput">
+      <TitleIcon style={{ paddingBottom: "180px", zIndex: 1 }} />
+      <textarea
+        ref={textareaRef}
+        disabled={isDisabled}
+        value={displayValue}
+        placeholder={placeholder}
+        onKeyDown={handleKeyDown}
+        onSelect={handleSelectionChange}
+        style={{
+          width: "200px",
+          height: "200px",
+          marginLeft: "-25px",
+          textIndent: "25px",
+          lineHeight: "30px",
+          color: isValid ? "green" : "red",
+          animation: isValid ? "none" : "shake 0.75s ",
+          resize: "none",
+        }}
+      />
+      <KeyboardReturnIcon
+        style={{
+          marginLeft: "-30px",
+          marginTop: "180px",
+          color: submitHover ? "lightblue" : "black",
+          transform: submitMouseDown ? "scale(0.9)" : "scale(1)",
+        }}
+        onClick={isDisabled ? () => {} : () => console.log(realValue)}
+        onMouseEnter={isDisabled ? () => {} : () => setSubmitHover(true)}
+        onMouseLeave={isDisabled ? () => {} : () => setSubmitHover(false)}
+        onMouseDown={isDisabled ? () => {} : () => setSubmitMouseDown(true)}
+        onMouseUp={isDisabled ? () => {} : () => setSubmitMouseDown(false)}
+      />
+      <InfoIcon
+        style={{
+          marginLeft: "-20px",
+          color: infoHover ? "lightblue" : "black",
+          transform: infoMouseDown ? "scale(0.9)" : "scale(1)",
+        }}
+        onClick={isDisabled ? () => {} : () => console.log("Info")}
```

Comment:
This function should also recieve a callback from the parent (add a prop)
--
I would expect following behavior from the info icon

```ts
<TextInput info={{ text: "only alphanumeric characters" }} />

if (!info) { hide info icon }
```